### PR TITLE
Feat/credential

### PR DESCRIPTION
**Pull request by BaskarMitrah**
## Description

Upgraded Credential Phase 2

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -172,5 +174,26 @@ module.exports = {
...
+      secure: false,
+      changeOrigin: true,
+    }));
+
+    app.use("/ims", createProxyMiddleware({
+      target: "https://developer-stage.adobe.com/",
+      secure: false,
+      changeOrigin: true,
+    }));
+  },
```
Comment (dc297) [inline gatsby-config.js:198]: These changes should not be merged.

File: package.json [hunk 2]
```diff
@@ -12,8 +12,9 @@
     "url": "https://github.com/icaraps"
   },
   "dependencies": {
-    "@adobe/gatsby-theme-aio": "^4.10.7",
+    "@adobe/gatsby-theme-aio": "^4.14.6",
     "gatsby": "4.22.0",
+    "http-proxy-middleware": "^2.0.6",
```
Comment (dc297) [inline package.json:17]: Let's remove this dependency

File: package.json [hunk 3]
```diff
@@ -22,6 +23,7 @@
     "start:prefix": "gatsby build --prefix-paths && gatsby serve --prefix-paths",
     "dev": "gatsby develop",
     "dev:https": "gatsby develop --https --host localhost.corp.adobe.com --port 9000",
+    "dev:https:local": "gatsby develop --https --host localhost.corp.adobe.com --port 8000",
```
Comment (dc297) [inline package.json:26]: Why is this needed?

File: src/components/CreateCredential.js [hunk 4]
```diff
@@ -1,69 +1,134 @@
...
-        }
-      }
-    }
-  `
-  );
+import creativeCloud from "../pages/assets/cc-icon.png";
+
+const GetCredentialApiKey = () => {
+
+  const { GATSBY_TEMPLATE_ID, GATSBY_PRODUCT_NAME } = process.env;
```
Comment (dc297) [inline src/components/CreateCredential.js:7]: Nice!

File: src/components/CreateCredential.js [hunk 5]
```diff
@@ -1,69 +1,134 @@
...
+        <GetCredential.Form.Products label="Included products and services">
+          <GetCredential.Form.Product label="Adobe Express Embed SDK" icon={creativeCloud} />
+        </GetCredential.Form.Products>
 
         <GetCredential.Form.Downloads label="Download a personalized code sample" contextHelp={true} contextHelpHeading="Select Language">
-          <GetCredential.Form.Download title="JavaScript" href={data?.allFile?.edges[0]?.node?.publicURL} />
+          <GetCredential.Form.Download title="JavaScript" href="https://acrobatservices.adobe.com/dc-integration-creation-app-cdn/8bab684/files/samples_q3_2023/PROD/dc-pdf-services-sdk-java-samples.zip" />
```
Comment (dc297) [inline src/components/CreateCredential.js:25]: The href here is wrong. It points to acrobat services zip. We should keep using the old href `data?.allFile?.edges[0]?.node?.publicURL`.

File: src/components/CreateCredential.js [hunk 6]
```diff
@@ -1,69 +1,134 @@
...
-            <a className='side-documentation' href='https://developer.adobe.com/express/embed-sdk/docs/guides'>Adobe Express Embed SDK documentation</a>
+            <a className="side-documentation" style={{ color: "#0265DC" }} href='https://developer.adobe.com/express/embed-sdk/docs/guides/quickstart/'>Quickstart guide</a>
+            <a className='side-documentation' style={{ color: "#0265DC" }} href='https://developer.adobe.com/express/embed-sdk/docs/guides/'>Adobe Express Embed SDK documentation</a>
           </div>
         </GetCredential.Form.Side>
 
       </GetCredential.Form>
 
-      <GetCredential.Card title="Your credential is ready to use" paragraph="Check the downloads section of your browser for the ZIP file, or find it where you save downloads on your machine." nextStepsLabel="Next steps" nextStepsHref="/credentials/nextsteps" devConsoleDirection="project_overview|api_overview|credential_overview" developerConsoleManage="Manage on Developer console" className="card_developer_console">
+      <GetCredential.UnknownError helpLinkText="Get Help" className="unKnownError" />
```
Comment (dc297) [inline src/components/CreateCredential.js:42]: Is there a `helpLinkHref` attribute for this element?

File: src/components/CreateCredential.js [hunk 7]
```diff
@@ -1,69 +1,134 @@
...
 
-        <GetCredential.Side>
-          <div className='side-container'>
+      <GetCredential.Card title="Your credential is ready to use" paragraph="Check the downloads section of your browser for the ZIP file, or find it where you save downloads on your machine." nextStepsLabel="Next steps" nextStepsHref="/credentials/nextsteps" className="card_developer_console">
```
Comment (dc297) [inline src/components/CreateCredential.js:44]: Does this `nextStepsHref` exist?

**Review by dc297 — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (ErinFinnegan) [reply package.json:26]: 🤔 I didn't add this...?

Comment (ErinFinnegan) [reply package.json:26]: Woops, I've been looking at the wrong PR... 😓

File: gatsby-config.js (hunk 1 above)
Comment (BaskarMitrah) [reply gatsby-config.js:198]: Okay @dc297

File: package.json (hunk 2 above)
Comment (BaskarMitrah) [reply package.json:17]: Okay

File: package.json (hunk 3 above)
Comment (BaskarMitrah) [reply package.json:26]: For local we use this port to run the application that's why I'm adding this. Okay I'll remove

File: src/components/CreateCredential.js (hunk 5 above)
Comment (BaskarMitrah) [reply src/components/CreateCredential.js:25]: Okay Sure, I'll change

File: src/components/CreateCredential.js (hunk 7 above)
Comment (BaskarMitrah) [reply src/components/CreateCredential.js:44]: Yes Deepesh.

File: src/components/CreateCredential.js (hunk 4 above)
Comment (timkim) [reply src/components/CreateCredential.js:7]: Do these env vars need to be added to the `deploy.yml`? 

https://github.com/AdobeDocs/cc-everywhere/blob/main/.github/workflows/deploy.yml#L153

File: src/components/CreateCredential.js [hunk 8]
```diff
@@ -17,53 +20,131 @@ const CreateCredential = () => {
...
+
+        <GetCredential.Card.CredentialDetails heading="Credential details">
+          <GetCredential.Card.CredentialDetails.APIKey heading="API Key" />
+          <GetCredential.Card.CredentialDetails.AllowedOrigins heading="Allowed domains" />
+          <GetCredential.Card.CredentialDetails.OrganizationName heading="Organization" />
+        </GetCredential.Card.CredentialDetails>
 
       </GetCredential.Card>
 
+      <GetCredential.Return title="Previously created projects" paragraph="Select a project and access your existing credentials for Firefly - Firefly and Creative Cloud Automation." nextStepsLabel="Next steps" nextStepsHref="/credentials/nextsteps" className="card_developer_console">
```
Comment (deepessh) [inline src/components/CreateCredential.js:86]: Shouldn't it say `Adobe Express Embed SDK` instead of `Firefly - Firefly and Creative Cloud Automation`?

File: src/components/CreateCredential.js [hunk 9]
```diff
@@ -17,53 +20,131 @@ const CreateCredential = () => {
...
+
+        <GetCredential.Return.Side>
+          <GetCredential.Return.Side.Custom>
+            <div style={{ display: "flex", gap: "30px", flexDirection: "column", width: "100%" }}>
+              <h3 className='spectrum-Heading spectrum-Heading--sizeM'>Welcome back</h3>
+              <p className="spectrum-Body spectrum-Body--sizeM">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation.</p>
```
Comment (deepessh) [inline src/components/CreateCredential.js:92]: We need to replace this placeholder text

File: src/components/CreateCredential.js [hunk 10]
```diff
@@ -17,53 +20,131 @@ const CreateCredential = () => {
...
+      </GetCredential.Return>
+
+      <GetCredential.RequestAccess
+        title="Get credentials"
+        paragraph="Create unique credentials that you will use to call multiple APIs from your application."
+      >
+        <GetCredential.RequestAccess.EdgeCase>
+          <GetCredential.RequestAccess.EdgeCase.NoProduct title="Your organization does not have access to Adobe Express Embed SDK" buttonLabel="Contact us to learn more" buttonLink="#someLink" />
+          <GetCredential.RequestAccess.EdgeCase.NotMember title="Access to Adobe Express Embed SDK APIs is not available at this time." buttonLabel="Learn more about Adobe Express Embed SDK" buttonLink="#someLink" />
+          <GetCredential.RequestAccess.EdgeCase.NotSignUp title="Adobe Express Embed SDK APIs is available as part of the beta program. Sign up for the program or log in to an account that has access." buttonLabel="Sign up for the beta" buttonLink="#someLink" />
```
Comment (deepessh) [inline src/components/CreateCredential.js:123]: This case is not applicable for express embed sdk. Can we remove this?

File: src/components/CreateCredential.js [hunk 11]
```diff
@@ -17,53 +20,131 @@ const CreateCredential = () => {
...
+          buttonLabel="Request access"
+        >
+          <GetCredential.RequestAccess.RestrictedAccess.Products label="Included products and services">
+            <GetCredential.RequestAccess.RestrictedAccess.Products.Product label="Adobe Express Embed SDK" icon={creativeCloud} />
+          </GetCredential.RequestAccess.RestrictedAccess.Products>
+        </GetCredential.RequestAccess.RestrictedAccess>
+        <GetCredential.RequestAccess.RequestAccessSide>
+          <div style="display : flex ; gap : 16px ; flex-direction : column;">
+            <h3 className="spectrum-Heading spectrum-Heading--sizeS side-header" >API key credential</h3>
+            <p className="spectrum-Body spectrum-Body--sizeM">An API Key credential was created. The API key credential identifies your application to Adobe servers and can help accept or reject request originating from certain domains.</p>
```
Comment (deepessh) [inline src/components/CreateCredential.js:137]: Why does it say `An API Key credential was created.`?

**Review by deepessh — CHANGES_REQUESTED** [review]

File: src/components/CreateCredential.js (hunk 4 above)
Comment (BaskarMitrah) [reply src/components/CreateCredential.js:8]: yes

File: src/components/CreateCredential.js (hunk 8 above)
Comment (BaskarMitrah) [reply src/components/CreateCredential.js:86]: My bad , I'll update

File: src/components/CreateCredential.js (hunk 9 above)
Comment (BaskarMitrah) [reply src/components/CreateCredential.js:92]: Yes

File: src/components/CreateCredential.js (hunk 10 above)
Comment (BaskarMitrah) [reply src/components/CreateCredential.js:123]: Yeah , Sure

File: src/components/CreateCredential.js (hunk 9 above)
Comment (BaskarMitrah) [reply src/components/CreateCredential.js:92]: @ErinFinnegan, Could you please share the placeholder ?

File: src/components/CreateCredential.js (hunk 11 above)
Comment (BaskarMitrah) [reply src/components/CreateCredential.js:137]: @ErinFinnegan ,For this too

File: src/components/CreateCredential.js [hunk 12]
```diff
@@ -17,53 +20,116 @@ const CreateCredential = () => {
...
+  )
 
   return (
+    <GetCredential className="getCredentialContainer" templateId={GATSBY_TEMPLATE_ID} productName={GATSBY_PRODUCT_NAME} >
 
-    <GetCredential credentialType="apiKey" className="getCredentialContainer" service="CCEmbedCompanionAPI">
+      <GetCredential.SignIn title="Get credentials" paragraph="Create unique credentials that you will use to call multiple APIs from your application." buttonText="Sign in to create credentials" />
 
-      <GetCredential.SignIn title="Get credentials" paragraph="Create unique credentials that you will use to call the Adobe Express Embed SDK from your application." buttonText="Sign in to create credentials" className="SignInClass" />
+      <GetCredential.Form title="Get credentials" paragraph="Create unique credentials that you will use to call multiple APIs from your application." className="formClass">
```
Comment (deepessh) [inline src/components/CreateCredential.js:30]: Instead of multiple APIs, shouldn't this say `Adobe Express Embed SDK`? Since we only have one API in this case.

File: src/components/CreateCredential.js [hunk 13]
```diff
@@ -17,53 +20,116 @@ const CreateCredential = () => {
...
     }
   `
-  );
+  )
 
   return (
+    <GetCredential className="getCredentialContainer" templateId={GATSBY_TEMPLATE_ID} productName={GATSBY_PRODUCT_NAME} >
 
-    <GetCredential credentialType="apiKey" className="getCredentialContainer" service="CCEmbedCompanionAPI">
+      <GetCredential.SignIn title="Get credentials" paragraph="Create unique credentials that you will use to call multiple APIs from your application." buttonText="Sign in to create credentials" />
```
Comment (deepessh) [inline src/components/CreateCredential.js:28]: Instead of multiple APIs, shouldn't this say Adobe Express Embed SDK? Since we only have one API in this case.

**Review by deepessh — APPROVED** [review]

**Review by ErinFinnegan — APPROVED** [review]
lgtm
